connection_generator.py

```diff
@@ -0,0 +1,197 @@
+
+        get_many_to_many_connections()
+            Creates connections between two groups with many to many relationship.
+
+        _dataset_to_dataset_collection()
+            Generates dataset - dataset collection connections.
+
+        _system_to_system_collection()
+            Generates system - system collection connections.
+
+        _dataset_read_to_system_input()
+            Generates connections between dataset reads and system inputs.
+
+        _dataset_write_to_system_output()
+            Generates connections between dataset write and system outputs.
+
+        generate()
+            Generates all the needed connections for data dependency mapping graph.
+    """
+    def __init__(self, dataset_count, dataset_count_map, system_count, system_count_map, dataset_read_count,
+                 system_input_count, dataset_write_count, system_output_count, dataset_read_count_map,
+                 dataset_write_count_map, system_input_count_map, system_output_count_map):
+        self.dataset_count = dataset_count
+        self.dataset_count_map = dataset_count_map
+
+        self.system_count = system_count
+        self.system_count_map = system_count_map
+
+        self.dataset_read_count = dataset_read_count
+        self.dataset_write_count = dataset_write_count
+        self.system_input_count = system_input_count
+        self.system_output_count = system_output_count
+        self.dataset_read_count_map = dataset_read_count_map
+        self.system_input_count_map = system_input_count_map
+        self.dataset_write_count_map = dataset_write_count_map
+        self.system_output_count_map = system_output_count_map
+
+        self.datasets_conn_collection = {}
+        self.systems_conn_collection = {}
+        self.dataset_read_conn_systems = {}
+        self.dataset_write_conn_systems = {}
+
+    @staticmethod
+    def get_one_to_many_connections(element_count, element_count_map):
+        """Generate group id for each element, based on number of element in group distribution.
+
+        Args:
+            element_count: Total number of elements.
+            element_count_map: Dictionary that maps element count in a group to number of groups with that count.
+
+        Returns:
+            Dictionary that maps group id to a list of element ids.
+        """
+        # Create element ids
+        element_values = list(range(1, element_count + 1))
+
+        # Get number of elements for each group id from their count
+        elements_per_group = [i for i in element_count_map for _ in range(element_count_map[i])]
+
+        # Randomise element ids and group ids
+        random.shuffle(element_values)
+        random.shuffle(elements_per_group)
+
+        it = iter(element_values)
+
+        # Split element ids into chunks to get connections for each group
+        group_to_elements = {i + 1: set(islice(it, 0, elements_per_group[i])) for i in range(len(elements_per_group))}
```

Comment:
Does this need to be an iter?

Reply:
It doesn't have to be an iter, could be a loop with list slicing and setting it as a value in a dictionary. This way just looks prettier to me.

Reply:
I mean "it", does it need to be an iter (rather than a list)?